### ISSUE TITLE
Plan for Java/Node SDK?

### ISSUE THREAD

**Issue by djKooks** [indradb/indradb#229]
Hello,
Seems there are SDK for python. Don't you have plan for Java, or Node?

**Comment by ysimonson** [comment]
No plans, but happy to help anyone who wants to undertake the effort. Since it's all gRPC, building bindings would be straight-forward -- potentially no code would have to actually be written.

**Comment by djKooks** [comment]
@ysimonson yes, I've saw Python SDK, and it seems pretty simple to go on. Let me look around 🙏

**Comment by djKooks** [comment]
@ysimonson hello
I'm reviewing python-sdk to refer implementation. But it has been blocked as following
```
$ make init
...
fatal: clone of 'git@github.com:indradb/indradb.git' into submodule path '/Users/kination/workspace/oss/python-client/indradb_server' failed
Failed to clone 'indradb_server'. Retry scheduled
Cloning into '/Users/kination/workspace/oss/python-client/indradb_server'...
Enter passphrase for key '/Users/kination/.ssh/id_rsa': 
```

could you check this?

**Comment by ysimonson** [comment]
It's cloning a submodule, and asking for the passphrase to the SSH key that you use for git.

**Comment by djKooks** [comment]
@ysimonson oh, sorry. It was issue in my laptop. Thanks for quick feedback 🙏

**Comment by kination** [comment]
@ysimonson hello.
Bit a long time, and started to work on node SDK.

There are some question, that when I call `createVertex` API, it always returns
```
{
      wrappers_: null,
      messageId_: undefined,
      arrayIndexOffset_: -1,
      array: [],
      pivot_: 1.7976931348623157e+308,
      convertedPrimitiveFields_: {}
}
```
Could you let me know whether it is correct or not?

**Comment by kination** [comment]
There were sometimes returning expected value, such as returning `true` like below
```
{
      wrappers_: null,
      messageId_: undefined,
      arrayIndexOffset_: -1,
      array: [ true ],
      pivot_: 1.7976931348623157e+308,
      convertedPrimitiveFields_: {}
}
```

but most of cases it returns empty array data.
So my question is...

- How can I know if `CreateVertex` fails?
- How should I setup `id` for this? When I setup value as
```
const vert: IVertex = {
  id: "1234567890123456",
  identifier: "test"
}
```
it shows => `uuid conversion failed: invalid length: expected 16 bytes, found 12`
- Is this returns failure when using same id to create?

**Comment by ysimonson** [comment]
That error is happening because you’re not passing a valid UUID (as a 16-byte array of bytes) for the id. I’m not sure about the rest, it looks like the gRPC implementation you’re using is doing something weird, giving back some low level details rather than the properly deserialized response.

**Comment by kination** [comment]
@ysimonson 
Thanks for feedback. But when I send proper 16-byte uuid, in most of cases it returns the following format (empty array)
https://github.com/indradb/indradb/issues/229#issuecomment-1565158483

As I think it should return true/false inside 'array'.

How can I debug it?

**Comment by kination** [comment]
I've checked several few things, and seems when...
- With appropriate uuid, which is unique, it returns `true`
- With appropriate uuid, which is already exists in db, it returns empty value
- With non-appropriate uuid, it cause exception.

**Comment by kination** [comment]
Not quite sure, but seems returning empty array is just happening that node indicates array with false as empty one.
So I think this issue is okay.

But for the future, it will be good if returning data includes detail log 🤔
